src/main/java/Parser.java

```diff
@@ -0,0 +1,94 @@
+                            ui.printDone(tasklist.get(index - 1));
+                            storage.save(tasklist);
+                            input = ui.readCommand();
+                            break;
+                        case "todo":
+                            String tododescription = input.substring(5);
+                            Task todotask = new ToDo(tododescription);
+                            TaskList.add(todotask, numoftasks);
+
+                            numoftasks++;
+                            ui.echo(tasklist.get(numoftasks - 1), numoftasks);
+                            storage.save(tasklist);
+                            input = ui.readCommand();
+                            break;
+                        case "event":
+                            String eventdescription = input.split(" /at ")[0].substring(6);
+                            String at = input.split(" /at ")[1];
+                            if (at.contains("/") && at.contains(" ")) {
+                                at = DateReader.readDate(new DateReader(at));
+                            } else {}
+
+                            Event eventtask = new Event(eventdescription, at);
+                            TaskList.add(eventtask, numoftasks);
+
+                            numoftasks++;
+                            ui.echo(tasklist.get(numoftasks - 1), numoftasks);
+                            storage.save(tasklist);
+                            input = ui.readCommand();
+                            break;
+                        case "deadline":
+                            String deadlinedescription = input.split(" /by ")[0].substring(9);
+                            String by = input.split(" /by ")[1];
+                            if (by.contains("/") && by.contains(" ")) {
+                                by = DateReader.readDate(new DateReader(by));
+                            } else {}
+
+                            Deadline deadlinetask = new Deadline(deadlinedescription, by);
+                            TaskList.add(deadlinetask, numoftasks);
+
+                            numoftasks++;
+                            ui.echo(tasklist.get(numoftasks - 1), numoftasks);
+                            storage.save(tasklist);
+                            input = ui.readCommand();
+                            break;
+                        case "delete":
+                            if (tasklist.get(0) == null) {
+                                throw new TaskDoesNotExistException("That task does not exist!");
+                            } else if (tasklist.get(Integer.parseInt(input.substring(7)) - 1) == null) {
+                                throw new TaskDoesNotExistException("That task does not exist!");
+                            } else {
+                                int indextodel = Integer.parseInt(input.substring(7));
+                                Task removed = tasklist.delete(indextodel);
```

Comment:
this class can be broken down further, long implementations and very long functions violate style